### PR TITLE
Improve Docker docs for use with Postgres

### PR DESCRIPTION
**Pull request by chmac**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [ ] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [ ] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Couple of tweaks to documentation. Doesn't include all the above methods, assuming they don't all apply to docs.

Partly this came from #8304.

### CODE REVIEW

File: contrib/docker/docker-compose.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ services:
     # failure
     restart: unless-stopped
     # See the readme for a full documentation of the environment settings
+    # NOTE: You must edit the config file to use postgres, it defaults to sqlite
```
Comment (DMRobertson) [inline contrib/docker/docker-compose.yml:17]:
```suggestion
    # NOTE: You must edit homeserver.yaml to use postgres, it defaults to sqlite
```

**Review by DMRobertson — COMMENTED** [review]
Thanks for this! For CI to pass, I think we just need a newsfile entry (`11640.doc`) which describes the change.

(Given that we don't recommend sqlite for production, I wonder if we ought to make the official docker images default to using postgres? cc @matrix-org/synapse-core )

**Review by DMRobertson — CHANGES_REQUESTED** [review]
as above, newsfile please!

**Comment by chmac** [comment]
@DMRobertson Switching the default on docker would make a lot of sense.

I'm not sure what a news file is or how to add one. Also, I won't make any more changes to the PR. I'm happy for somebody else to do whatever steps are required or abandon it altogether, I have no attachment to it getting merged. I thought it might make an improvement to the docs, but I leave that up to others to decide.

Happy to close this if it requires work to get merged.

File: docker/README.md [hunk 2]
```diff
@@ -68,6 +68,10 @@ The following environment variables are supported in `generate` mode:
   directories. If unset, and no user is set via `docker run --user`, defaults
   to `991`, `991`.
 
+## Postgres
+
+By default the config will use sqlite. See the [docs on using postgres](https://github.com/matrix-org/synapse/blob/develop/docs/postgres.md) for more info on how to use postgres. Until this section is improved [this issue](https://github.com/matrix-org/synapse/issues/8304) may provide useful information.
```
Comment (reivilibre) [inline docker/README.md:73]:
```suggestion
By default the config will use SQLite. See the [docs on using Postgres](https://github.com/matrix-org/synapse/blob/develop/docs/postgres.md) for more info on how to use Postgres. Until this section is improved [this issue](https://github.com/matrix-org/synapse/issues/8304) may provide useful information.
```

**Pull request merged**